src/bot/commands/command.ts

```diff
@@ -15,6 +21,28 @@ export abstract class Command {
 
   abstract execute(message: Message, input?: string): Promise<void>;
 
+  async getAudioResource(url: string) {
+    const source = this.getSourceStream();
+
+    const stream = await source.getStream(url);
+
+    return createAudioResource(stream, {
+      inputType: StreamType.Opus,
+    });
+  }
+
+  async getResourceUrl(title: string, artist: string) {
+    const source = this.getSourceStream();
+
+    const search = (
+      await source.search(`${title} ${artist}`, {
+        limit: 1,
+      })
+    )[0] as ResultAudioSearch;
```

Comment:
You can use shift here as well to get first item

Reply:
resolved: [5252756](https://github.com/Marli-Marlete/marli-music/pull/21/commits/5252756b4ba49d970306720a3bb9a66014cda2c6)